model/websocket_client_test.go

```diff
@@ -149,3 +150,53 @@ func TestWebSocketClose(t *testing.T) {
 		checkWriteChan(cli.writeChan)
 	})
 }
+
+func binaryWebsocketHandler(t *testing.T, clientData map[string]interface{}) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		upgrader := &websocket.Upgrader{
+			ReadBufferSize:  1024,
+			WriteBufferSize: 1024,
+		}
+		conn, err := upgrader.Upgrade(w, req, nil)
+		require.NoError(t, err)
+		defer conn.Close()
+
+		for {
+			msgType, buf, err := conn.ReadMessage()
+			require.NoError(t, err)
+			if msgType == websocket.BinaryMessage {
+				require.Equal(t, msgType, websocket.BinaryMessage)
+				wsReq := &WebSocketRequest{}
+				err = msgpack.Unmarshal(buf, wsReq)
+				require.NoError(t, err)
+				require.Equal(t, clientData, wsReq.Data)
+				break
+			}
+		}
+	}
+}
+
+func TestWebSocketSendBinaryMessage(t *testing.T) {
+	clientData := map[string]interface{}{
+		"data": []byte("some data to send as binary"),
+	}
+
+	s := httptest.NewServer(binaryWebsocketHandler(t, clientData))
+	defer s.Close()
+
+	url := strings.Replace(s.URL, "http://", "ws://", 1)
+	cli, err := NewWebSocketClient4(url, "authToken")
+	require.NoError(t, err)
+
+	cli.Listen()
+
+	err = cli.SendBinaryMessage("binaryAction", map[string]interface{}{
+		"func": func() {},
+	})
+	require.Error(t, err)
+
+	err = cli.SendBinaryMessage("binaryAction", clientData)
+	require.NoError(t, err)
+
+	<-cli.writeChan
```

Comment:
Can you clarify why this is needed? Also, shouldn't we do `cli.Close()`?

Reply:
I was trying to make sure that the message was handled prior to exiting the test. The close happens on the server side handler upon reception of the binary message. I've updated this now.